src/modules/dashboard/components/ColonyHome/ColonyDomainSelector/ColonyDomainSelector.tsx

```diff
@@ -0,0 +1,121 @@
+import { useEnabledExtensions } from '~utils/hooks/useEnabledExtensions';
+
+import CreateDomainButton from './CreateDomainButton';
+
+import styles from './ColonyDomainSelector.css';
+
+interface FormValues {
+  filteredDomainId: string;
+}
+
+interface Props {
+  filteredDomainId?: number;
+  onDomainChange?: (domainId: number) => any;
+  colony: Colony;
+}
+
+const displayName = 'dashboard.ColonyHome.ColonyColonyDomainSelector';
+
+const ColonyDomainSelector = ({
+  filteredDomainId = COLONY_TOTAL_BALANCE_DOMAIN_ID,
+  onDomainChange,
+  colony,
+}: Props) => {
+  const { networkId, ethereal, username } = useLoggedInUser();
+  const { isVotingExtensionEnabled } = useEnabledExtensions({
+    colonyAddress: colony.colonyAddress,
+  });
+
+  const openEditDialog = useDialog(EditDomainDialog);
+  const handleEditDomain = useCallback(
+    (ethDomainId: number) =>
+      openEditDialog({
+        selectedDomainId: String(ethDomainId),
+        colony,
+        isVotingExtensionEnabled,
+      }),
+    [openEditDialog, colony, isVotingExtensionEnabled],
+  );
+
+  const getDomainColor = useCallback<(domainId: string | undefined) => Color>(
+    (domainId) => {
+      const rootDomainColor: Color = Color.Yellow;
+      const defaultColor: Color = Color.LightPink;
+      if (domainId === String(ROOT_DOMAIN_ID)) {
+        return rootDomainColor;
+      }
+      if (!colony || !domainId) {
+        return defaultColor;
+      }
+      const domain = colony.domains.find(
+        ({ ethDomainId }) => Number(domainId) === ethDomainId,
+      );
+      return domain ? domain.color : defaultColor;
+    },
+    [colony],
+  );
+
+  const renderActiveOption = useCallback<
+    (option: SelectOption | undefined, label: string) => ReactNode
+  >(
+    (option, label) => {
+      const value = option ? option.value : undefined;
+      const color = getDomainColor(value);
+      return (
+        <div className={styles.activeItem}>
+          <ColorTag color={color} />{' '}
+          <div className={styles.activeItemLabel}>{label}</div>
+        </div>
+      );
+    },
+    [getDomainColor],
+  );
+
+  const isNetworkAllowed = !!ALLOWED_NETWORKS[networkId || 1];
+  const isSupportedColonyVersion =
+    parseInt(colony.version, 10) >= ColonyVersion.LightweightSpaceship;
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canInteract =
+    isSupportedColonyVersion &&
+    isNetworkAllowed &&
+    hasRegisteredProfile &&
+    colony?.isDeploymentFinished;
+
+  return (
+    <Form<FormValues>
+      initialValues={{
+        filteredDomainId: String(filteredDomainId),
+      }}
+      onSubmit={() => {}}
+    >
+      <DomainDropdown
+        colony={colony}
+        name="filteredDomainId"
+        currentDomainId={filteredDomainId}
+        onDomainChange={onDomainChange}
+        onDomainEdit={canInteract ? handleEditDomain : undefined}
+        footerComponent={
+          canInteract ? <CreateDomainButton colony={colony} /> : undefined
```

Comment:
This won't work due to the typing inside the component. It needs to return `undefined`, because your version returns `null` and that causes a type error.